subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/AbstractListChildMap.java

```diff
@@ -35,14 +36,16 @@ public boolean isEmpty() {
 
     @Override
     public List<T> values() {
-        return entries.stream()
-            .map(Entry::getValue)
-            .collect(Collectors.toList());
+        List<T> values = new ArrayList<>(entries.length);
+        for (Entry<T> entry : entries) {
+            values.add(entry.getValue());
+        }
+        return values;
```

Comment:
The only production code call to this in `DirectorySnapshot`'s constructor turns the result into a `Stream` immediately. We could return a `Stream` here instead. I think that would further reduce the garbage and increase speed.